src/ui/common/src/components/tables/PaginatedSearchTable.tsx

```diff
@@ -125,53 +153,196 @@ export const PaginatedSearchTable: React.FC<PaginatedSearchTableProps> = ({
     return value;
   };
 
-  if (searchQuery.length > 0) {
-    filteredRows = data.data.filter((rowItem) => {
-      return shouldInclude(rowItem, searchQuery, searchColumn);
+  React.useEffect(() => {
+    if (searchQuery.length > 0) {
+      const filteredRows = data.data.filter((rowItem) => {
+        return shouldInclude(rowItem, searchQuery, searchColumn);
+      });
+
+      setRows(filteredRows);
+    } else {
+      setRows(data.data);
+    }
+  }, [searchQuery]);
+
+  React.useEffect(() => {
+    if (
+      !sortConfig.sortColumn ||
+      !sortConfig.sortColumn.name ||
+      !sortConfig.sortType
+    ) {
+      setRows(data.data);
+      return;
+    }
+
+    const sortedRows = [...rows].sort((r1, r2) => {
+      const col = sortConfig.sortColumn;
+      let v1: PaginatedSearchTableRow | PaginatedSearchTableElement = r1;
+      let v2: PaginatedSearchTableRow | PaginatedSearchTableElement = r2;
+      for (const path of col.sortAccessPath) {
+        v1 = v1[path];
+        v2 = v2[path];
+      }
+
+      if (sortConfig.sortType === 'asc') {
```

Comment:
Kind of a silly thing, but can you add a const enum for the sort types? 
That way in the future, we can just say 'SORT_TYPE.ASC' and have stronger typing.

Reply:
Yep, makes sense!